tests/test_frame.py

```diff
@@ -1278,6 +1278,18 @@ def test_indexslice_setitem():
     df.loc[pd.IndexSlice[2, :], "z"] = [200, 300]
 
 
+def test_indexslice_getitem():
+    # GH 300
+    df = (
+        pd.DataFrame({"x": [1, 2, 2, 3, 4], "y": [10, 20, 30, 40, 10]})
+        .assign(z=lambda df: df.x * df.y)
+        .set_index(["x", "y"])
+    )
+    ind = pd.Index([2, 3])
+    check(assert_type(pd.IndexSlice[ind, :], tuple[pd.Index, slice]), tuple)
+    check(assert_type(df.loc[pd.IndexSlice[ind, :]], pd.DataFrame), pd.DataFrame)
```

Comment:
Ideally you would have a test that would show all possible types can be generated.

Reply:
Some of them are tested elsewhere, but adding more tests is a good idea.  Will work on that.

Reply:
I added a bunch of tests in the latest commit.